hw1/CBC-HMAC.py

```diff
@@ -51,15 +51,25 @@ def __strip_padding(self, data):
         :param data: input data
         :return: stripped data
         """
-        ?
+        pad_len = data[-1]
+        actual_data, padding = data[:-(pad_len + 1)], data[-(pad_len + 1): -1]
+        for byte in padding:
+            if byte != pad_len:
+                return None # invalid padding
+        expected_pad_len = self.block_len - ((len(actual_data) + 1) % self.block_len)
+        if expected_pad_len != pad_len:
+            return None # invalid padding
+        return actual_data
 
     def __pad(self, data):
         """
         Pad the data so that the block size is a multiple of block_len
         :param data: input data
         :return: padded data with length an integral multiple of block_len
         """
-        ?
+        pad_len = self.block_len - ((len(data) + 1) % self.block_len)
+        padding = pad_len.to_bytes(1, 'little') * (pad_len + 1)
```

Comment:
```suggestion
        padding = pad_len.to_bytes(1, byteorder='big') * (pad_len + 1)
```
Doesn't really matter but we do that everywhere